### DISCUSSION TITLE
Mesh-agent not able to connect due to trojan file in exe

### DISCUSSION THREAD

**Discussion by a-imran99** [Ylianst/MeshCentral#4609] [Q&A] [answered]
Hi, @Ylianst .. I was trying to install mesh agent in host machine but the machine is blocking the connection due to the presence of trojan file in meshagent exe .

This is a Winodws64 agent.

Would be great if you replied ! :) @Ylianst 

Thanks.

![MicrosoftTeams-image](https://user-images.githubusercontent.com/77147011/194374704-3675d8d9-a728-49e2-a6ed-55d236e29e9c.png)

**Comment by si458** [comment, marked as answer]
This looks like a false positive from Microsoft. You should be able to report it as a false virus and then run the exe anyways

**Reply by a-imran99** [reply]
Yes looks like a false positive.. 

But I am also looking for a way to build an exe out of mesh-agent code.

Is that possible? @si458

**Comment by PathfinderNetworks** [comment]
This is where customizing your agent and, especially important, acquiring a code signing certificate and signing your agents with that certificate.  In my case this was the only way I could get my agents to play nice with Avast CloudCare.  Since doing so (and after having them whitelist my signed agent) I've had no issues.  I found it takes a combination of the code signing, time stamping, and customizing the agent (including using your own logo in addition to naming it) to get around this.

**Reply by a-imran99** [reply]
Hi @PathfinderNetworks Can you provide any steps on how to build the mesh agent!. 

There is a video on how to customize the agent by @Ylianst . But that doesn't include how to sign the certificates.

it would be great if you told how to build the mesh agent. @PathfinderNetworks 

Thanks.

**Reply by si458** [reply]
@a-imran99 2 videos for code signing coming right up

https://www.youtube.com/watch?v=qMAestNgCwc

https://www.youtube.com/watch?v=xYtk85Bmc8Q&t=3s

Both from @Ylianst himself, checkout all his other videos on meshcentral for further help

**Reply by a-imran99** [reply]
Hey @si458 Thanks for the video. 

https://github.com/Ylianst/MeshAgent

I just need to make some changed to agent and then build the agent code. But I don't find resource for building this one. Help would be great.
